core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java

```diff
@@ -321,6 +321,20 @@ private static String toSql(RelNode root, SqlDialect dialect) {
         .ok(expectedMySql);
   }
 
+  @Test public void testSelectQueryWithGroupBySubQuery() {
+    String query = "select \"product_class_id\", count(\"product_id\")\n"
+        + "from (select \"product_class_id\", \"product_id\"\n"
+        + "from \"product\"\n"
+        + "group by \"product_class_id\", \"product_id\") as t\n"
+        + "group by \"product_class_id\"";
+    final String expected = "SELECT \"product_class_id\", COUNT(*)\n"
+        + "FROM (SELECT \"product_class_id\", \"product_id\"\n"
+        + "FROM \"foodmart\".\"product\"\n"
+        + "GROUP BY \"product_class_id\", \"product_id\") AS \"t0\"\n"
+        + "GROUP BY \"product_class_id\"";
+    sql(query).ok(expected);
+  }
```

Comment:
I can pass this test without the change in RelToSqlConverter.java and SqlImplementor.java

Reply:
Thanks,  the test case is not suitable, I will change it.